src/Particle/SampleStack.h

```diff
@@ -61,15 +56,13 @@ class SampleStack
   ///  Set the sample count to zero but preserve the storage
   void resetSampleCount();
 
-  ~SampleStack();
-
 private:
-  int total_num_;
-  int max_samples_;
-  int current_sample_count_;
-  uint64_t global_num_samples_;
+  int total_num_{0};
+  int max_samples_{10};
+  int current_sample_count_{0};
+  int64_t global_num_samples_{max_samples_};
```

Comment:
Can we change all of the above size_t?
size related integers should be always unsigned.

Don't like the default max_samples_ 0.

Reply:
> Don't like the default max_samples_ 0.

`max_samples_{10}` was in the default constructor. I'm happy to change it but don't know what a better value might be.

`global_num_samples_` was previously unset. Setting it to `max_samples_` may be incorrect, but I'm not sure we can always obtain `nranks` at construction.

Reply:
We need to document what max_samples_ does and use that info to decide what is a sensible value. We should also not change is just for the sake of changing it.

(More generally max anything in the code does not indicate high enough quality of design or implementation, but it is what we have.)

Reply:
@quantumsteve open an issue regarding this magic 10. No need to address here.

Reply:
MaxSamples and it's initialization to 10 was introduced in 2cba1257b989413ba5fbd6e85ab45e346e25cb15 ,
but it does not appear MaxSamples was used for anything in that commit.

Reply:
moved to issue #4497